File: client/di_core/DataIngestion/Domain/Response/DataSourceResponse.ts
import { DataSource, DataSourceInfo } from '@core/DataIngestion';
import { UserProfile } from '@core/domain';

export class DataSourceResponse {
  dataSource: DataSourceInfo;
  creator: UserProfile;

  constructor(dataSourceInfo: DataSourceInfo, creator: UserProfile) {
    this.dataSource = dataSourceInfo;
    this.creator = creator;
  }

  static fromObject(obj: any): DataSourceResponse {
    const dataSourceInfo = DataSourceInfo.fromObject(obj.dataSource);
    const creator = UserProfile.fromObject(obj.creator);
    return new DataSourceResponse(dataSourceInfo, creator);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobscheduler/domain/response/DataSourceResponse.scala
package co.datainsider.jobscheduler.domain.response

import co.datainsider.jobscheduler.domain.source.DataSource
import co.datainsider.caas.user_profile.domain.user.ShortUserProfile

case class DataSourceResponse(dataSource: DataSource, creator: Option[ShortUserProfile])
